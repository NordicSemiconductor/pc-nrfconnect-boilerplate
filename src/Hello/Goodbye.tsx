/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';

export default () => <h3>So long, and thanks for all the fish</h3>;
