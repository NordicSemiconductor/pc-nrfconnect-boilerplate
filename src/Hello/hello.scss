/*
 * Copyright (c) 2022 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

// This file is imported by Hello.jsx, and may contain styles for that component.

@import "~pc-nrfconnect-shared/styles";

.title {
    // Just an example
    border-left: $nordic-blue solid;
}
