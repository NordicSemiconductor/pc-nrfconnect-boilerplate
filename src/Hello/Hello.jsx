/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React, { useEffect } from 'react';
import { logger } from 'pc-nrfconnect-shared';
import { bool } from 'prop-types';

import './hello.scss';

const Hello = ({ active }) => {
    useEffect(() => {
        if (active) {
            logger.info('Showing Hello pane');
        }
        return () => {
            if (active) {
                logger.info('Hiding Hello pane');
            }
        };
    }, [active]);

    return <h3 className="title">Hello World</h3>;
};
Hello.propTypes = {
    active: bool,
};

export default Hello;
