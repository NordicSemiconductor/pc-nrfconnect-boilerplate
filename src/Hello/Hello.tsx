/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React, { useEffect } from 'react';
import { logger } from '@nordicsemiconductor/pc-nrfconnect-shared';

import './hello.scss';

const Hello: React.FC<{ active: boolean }> = ({ active }) => {
    useEffect(() => {
        if (active) {
            logger.info('Showing Hello pane');
        }
        return () => {
            if (active) {
                logger.info('Hiding Hello pane');
            }
        };
    }, [active]);

    return <h3 className="title">Hello World</h3>;
};

export default Hello;
