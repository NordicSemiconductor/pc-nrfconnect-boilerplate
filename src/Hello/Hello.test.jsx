/*
 * Copyright (c) 2015 Nordic Semiconductor ASA
 *
 * SPDX-License-Identifier: LicenseRef-Nordic-4-Clause
 */

import React from 'react';
import renderer from 'react-test-renderer';

import Hello from './Hello';

describe('Hello', () => {
    it('has a reliable rendering', () => {
        expect(renderer.create(<Hello />)).toMatchSnapshot();
    });
});
