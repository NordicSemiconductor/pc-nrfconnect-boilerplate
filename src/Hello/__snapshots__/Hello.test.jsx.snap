// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Hello should render component with custom title 1`] = `
<h3
  className="title"
>
  Test title
</h3>
`;

exports[`Hello should render component with default title 1`] = `
<h3
  className="title"
>
  default title
</h3>
`;
